### PR TITLE
ui-manchette/fix waypoint menu closing

### LINKED ISSUE #8629: Manchette : hide a waypoint by clicking on it
### Description and goal

In the manchette, we want to be able to hide a waypoint when clicking on it.

To work, this issue https://github.com/OpenRailAssociation/osrd-ui/issues/399 needs be done first.

### Acceptance criteria

- [x] When clicking on a waypoint, an options menu should be displayed
- [x] "Hide this waypoint" should be the only option visible for now
- [x] The menu's design should match the [sketch](https://www.sketch.com/s/a26d84be-5fb2-41e4-9e84-96d90f46a997/a/OeJV9yG) one 
- [x] Clicking on "Hide this waypoint" should close the menu and properly hide the waypoint
- [x] The space time chart should resize accordingly
- [x] The hidden waypoint should still be present in the waypoint list modal so it can't be displayed again
- [ ] When the menu is open, clicking on another waypoint or the same one just closes the menu and we need one more click to open another one
- [x] When the menu is open, the scroll in the manchette, the manchette toolbar and pan in space time chart are disabled
- [ ] If there was only two waypoints left, it shouldn't be possible to hide it (behavior needs to be checked with @thibautsailly)

### PR DESCRIPTION
**Pull request by SharglutDev**
See commits

close https://github.com/OpenRailAssociation/osrd/issues/8629 (see last comments for the issue)

### CODE REVIEW

File: ui-manchette-with-spacetimechart/src/stories/base-with-waypoint-menu.stories.tsx [hunk 1]
```diff
@@ -69,6 +77,36 @@ const ManchetteWithSpaceTimeWrapper = ({
...
+        setIsClickOnWaypoint(true);
+      }
+      // Close the menu if the user clicks outside of it
+      if (!menuRef.current?.contains(event.target as Node)) {
+        setActiveWaypointId(undefined);
+      }
+    };
+
+    const toggleButton = document.querySelector('.toggle-mode div');
+    const zoomButtons = document.querySelectorAll('.zoom-buttons span');
```
Comment (emersion) [inline ui-manchette-with-spacetimechart/src/stories/base-with-waypoint-menu.stories.tsx]: This is not safe to do in a `useEffect()`.

**Comment by emersion** [comment]
Hm, not a fan of these "clicked outside" event handlers with ad-hoc checks. Would prefer a transparent div covering the whole document (with a portal).

File: ui-manchette-with-spacetimechart/src/stories/base-with-waypoint-menu.stories.tsx (hunk 1 above)
Comment (SharglutDev) [reply ui-manchette-with-spacetimechart/src/stories/base-with-waypoint-menu.stories.tsx]: Layout then ? 😖

**Comment by SharglutDev** [comment]
> Hm, not a fan of these "clicked outside" event handlers with ad-hoc checks. Would prefer a transparent div covering the whole document (with a portal).

We need to discuss with @thibautsailly if having the waypoint menu displayed should block any first click interaction with the whole page (if yes we could maybe just use a modal ?) or just the manchette/spacetimechart area.

**Comment by SharglutDev** [comment]
Added the `createPortal` logic :)

**Review by emersion — APPROVED** [review]
LGTM :)

**Review by theocrsb — APPROVED** [review]
LGTM and tested

**Pull request merged**